### ISSUE TITLE
Improve display of long paths

### ISSUE THREAD

**Issue by simonmclean** [ThePrimeagen/harpoon#142]
The language I use at work is Scala. Being a JVM language, the directory structure tends to be deeply nested, and the file names often long. As a result the filename tends to get cut off in the harpoon window. 

Some ideas for how to remedy this

- Config option to display filenames only, without directories. Though this raises the question of how to differentiate files with the same name, but in different paths
- Config option for setting the width of the window
- Auto-expand window so that it fits contents
- Text wrapping
- Maybe a comination of auto-expand and text-wrap?

**Comment by sQVe** [comment]
Setting the width in the config is already on supported. Wrapping text is also working if you have it enabled already in neovim.

**Comment by simonmclean** [comment]
Cool, okay. Could the width option be documented? I'm not particularly knowledgable on lua or lua-based plugins, so it's not clear to me how to apply this.

Regarding text-wrap, personally I'd find it useful to have `wrap` set in the Harpoon menu, without changing my global `nowrap` setup. That's just me though. Don't know how many others would attach any importance to that.

**Comment by renerocksai** [comment]
> Regarding text-wrap, personally I'd find it useful to have `wrap` set in the Harpoon menu, without changing my global `nowrap` setup. That's just me though. Don't know how many others would find value in that.

The harpoon menu has the filetype `harpoon`. So you can define some autocmd to set wrap:

```vim
autocmd FileType harpoon set wrap
```

**Comment by simonmclean** [comment]
Thanks renerocksai! 👍

**Comment by sQVe** [comment]
I think `setlocal` is preferred when setting `wrap` via `autocmd`.

There's an menu option in the config. You can check my dots for how I setup harpoon: https://github.com/sQVe/dotfiles/blob/master/config/nvim/lua/plugins/harpoon.lua

**Comment by renerocksai** [comment]
Thanks. Just 2 things: good catch with `setlocal`.  I'll edit my post.

Yours read a bit misleading: after the `setlocal` stuff you continue with "There's a menu option in the config..." - which led me to believe there's a wrap option in the config that I then didn't find in your config.

**Comment by sQVe** [comment]
Yeah. My bad. Height and width is the only thing set via config, as far as I know.

**Comment by davvid** [comment]
@sQVe @simonmclean I added an example to the README in #163 that might be helpful. Maybe that's a good default value to consider. Is that example sufficient to close this issue?
